addons/hr_timesheet/wizard/hr_employee_delete_wizard.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+
+
+class HrEmployeDeleteWizard(models.TransientModel):
+    _name = 'hr.employee.delete.wizard'
+    _description = 'Employee Delete Wizard'
+
+    employee_count = fields.Integer(string='Employee Count', compute='_compute_employee_data')
+    employee_ids = fields.Many2many('hr.employee', string='Employees')
+    has_active_employee = fields.Boolean(string='Has Active Employee', compute='_compute_employee_data')
+    has_timesheet = fields.Boolean(string='Has Timesheet', compute='_compute_employee_data')
+
+    def _compute_employee_data(self):
+        self = self.with_context(active_test=False)
+        timesheet_read_group = self.env['account.analytic.line']._read_group([
+            ('employee_id', 'in', self.employee_ids.ids)],
+            ['employee_id'], ['employee_id']
+        )
+        timesheet_employee_map = set(employee_info['employee_id'][0] for employee_info in timesheet_read_group)
+        for wizard in self:
+            wizard.employee_count = len(wizard.employee_ids)
+            wizard.has_active_employee = wizard.with_context(active_test=True).employee_ids
+            wizard.has_timesheet = timesheet_employee_map & set(wizard.employee_ids.ids)
```

Comment:
I think it is a good practice to have one compute method for each field.
This way, when you get self.employee_count (e.g. in `action_open_timesheets()`), you don't need to calculate `has_timesheet` and so you avoid doing a _read_group. :)
Also, is a field necessary for employee_count?